### PR TITLE
Refine storage id

### PR DESCRIPTION
**Pull request by pilagod**
* Add id to account & network in storage
* Refine storage hooks

### CODE REVIEW

File: app/hook/storage.ts [hunk 1]
```diff
@@ -5,96 +5,65 @@ import { useStorage } from "~app/storage"
...
-  )
+export const useAccount = (id?: string) => {
+  const network = useNetwork()
+  return useStorage(({ state }) => {
+    return state.account[id ?? network.accountActive]
+  })
 }
 
-export const useAccount = (id?: string) => {
+export const useAccountCount = (networkId?: string) => {
```
Comment (WIN-LIN) [inline app/hook/storage.ts:32]: How about we check how many accounts in `useAccounts` directly? 
It seems that both of them are checking total accounts, and the diff is that `useAccountCount` can return the number of accounts.

Comment (pilagod) [reply app/hook/storage.ts:32]: `useAccounts` would re-render more frequently. Given now we don't encounter performance issue, I think it is a fine choice 👍

Comment (WIN-LIN) [reply app/hook/storage.ts]: I see. Thanks for the explanation!

**Review by WIN-LIN — APPROVED** [review]

**Comment by oneleo** [comment]
Summary:
1. Remove `useShallow` optimization for better readability.
2. Replace `useShouldOnboard()` with `useAccounts().length === 0`.
3. Copy network `id` into each network element's `id`.

**Review by oneleo — APPROVED** [review]

**Pull request merged**